python/pyrogue/utilities/fileio/_FileReader.py

```diff
@@ -16,74 +16,116 @@
 # copied, modified, propagated, or distributed except according to the terms 
 # contained in the LICENSE.txt file.
 #-----------------------------------------------------------------------------
-import rogue.utilities
-import rogue.utilities.fileio
 import pyrogue
 import rogue
 
+import os
+import struct
+import numpy
+from collections import namedtuple
 
-class FileReader(object):
+RogueHeaderSize  = 8
+RogueHeaderPack  = 'IHBB'
 
-    def __init__(self, filename, configChan=None):
-        self._filename   = filename
-        self._configChan = configChan
+# Default header as a named tuple
+RogueHeader = namedtuple( 'RogueHeader',
+                         [ 'size'                ,  # 4 Bytes, uint32, I
```

Comment:
Does size of 32-bit (4 bytes) means there a 4GB limit for the FileReader/FileWriter?  How does the FileReader/FileWriter handle the > 4GB case?

Reply:
The 32-bit size field is each record. I hope we don't have records that large.

Reply:
Does the FileWriter close the file and stop when the 4GB limit is reached?